cora/util/cubicspline.pyx

```diff
@@ -182,28 +181,40 @@ cdef class Interpolater(object):
         return (a * data[2*kl+1] + b * data[2*kh+1]
                 + c * y2[kl] + d * y2[kh])
 
-
+    @cython.boundscheck(False)  # Deactivate bounds checking
+    @cython.wraparound(False)   # Deactivate negative indexing.
     def __gen_spline_(self):
         # Solve for the second derivative matrix to generate the
         # splines.
 
-        n = self.__data_.shape[0]
-
-        al = np.zeros(n-2)
-        bt = np.zeros(n-2)
-        gm = np.zeros(n-2)
-        f = np.zeros(n-2)
-        z = np.zeros(n-2)
-        l = np.zeros(n-2)
-        m = np.zeros(n-2)
-
-        self.__y2_ = np.zeros(n)
+        cdef Py_ssize_t n = self.__data_.shape[0]
+        cdef Py_ssize_t length = n - 2
+        cdef Py_ssize_t i
+
+        al_raw = np.zeros(length, dtype=np.float64)
+        bt_raw = np.zeros(length, dtype=np.float64)
+        gm_raw = np.zeros(length, dtype=np.float64)
+        f_raw = np.zeros(length, dtype=np.float64)
+        z_raw = np.zeros(length, dtype=np.float64)
+        l_raw = np.zeros(length, dtype=np.float64)
+        m_raw = np.zeros(length, dtype=np.float64)
+
+        cdef double[:] al = al_raw
+        cdef double[:] bt = bt_raw
+        cdef double[:] gm = gm_raw
+        cdef double[:] f = f_raw 
+        cdef double[:] z = z_raw 
+        cdef double[:] l = l_raw 
+        cdef double[:] m = m_raw 
```

Comment:
Why do this in two steps? I think you can just assign the array to the array views on a single line.

Also for internal arrays like this you're apparently better off using a `cython.array` rather than `numpy.ndarray`, although it probably doesn't make much difference.